ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTree.java

```diff
@@ -0,0 +1,164 @@
+package us.ihmc.behaviors.behaviorTree;
+
+import behavior_msgs.msg.dds.BehaviorTreeStateMessage;
+import org.apache.commons.lang3.mutable.MutableLong;
+import us.ihmc.behaviors.behaviorTree.topology.BehaviorTreeTopologyOperationQueue;
+import us.ihmc.communication.crdt.CRDTInfo;
+import us.ihmc.communication.crdt.LatestTimestampModifiable;
+import us.ihmc.communication.ros2.ROS2ActorDesignation;
+import us.ihmc.communication.ros2.sync.ROS2PeerClockOffsetEstimator;
+import us.ihmc.tools.io.WorkspaceResourceDirectory;
+
+import java.util.function.Consumer;
+
+/**
+ * Common code for managing the tree between high layer implementations. i.e. RDX or Executor.
+ *
+ * @param <HLT> The generic type of this node high layer: RDX or Executor
+ */
+public abstract class BehaviorTree<HLT extends BehaviorTreeNodeHighLayer<HLT, ? ,?>>
+{
+   private int numberOfNodes = 0;
+   private final CRDTInfo crdtInfo;
+   private final MutableLong nextID = new MutableLong(0);
+   private final LatestTimestampModifiable rootReferenceModification;
+   private final LatestTimestampModifiable dataModification;
```

Comment:
data & root reference are now separate modifiable CRDT fields. That way the topology is synced separately from the data contained by the tree and nodes.